opentelemetry-sdk/src/opentelemetry/sdk/metrics/export/aggregate.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import abc
+import threading
+
+
+class Aggregator(abc.ABC):
+    """Base class for aggregators.
+
+    Aggregators are responsible for holding aggregated values and taking a
+    snapshot of these values upon export (check_point).
+    """
+
+    def __init__(self):
+        self.current = None
+        self.check_point = None
+        self._lock = threading.Lock()
```

Comment:
is a lock the right thing to put in the base class? I understand most implementations will want that, but it's not particularly related to the interface, and we may be needlessly locking if we have an async aggregator or use atomic integers and doubles.

Reply:
I'm not sure if an async aggregator makes sense, as we would want to export metric values at a certain point in time. With that being said, all implementations would need a way to handle concurrent updates, so I would think putting the lock in the base class makes sense.